suite-common/fiat-services/src/coingecko.ts

```diff
@@ -103,39 +103,66 @@ export const fetchCurrentFiatRates = async (ticker: TickerId) => {
 };
 
 /**
- * Returns the historical rate for a given symbol, timestamp fetched from CoinGecko API.
- * Be aware that the data granularity is 1 day.
- * Returns null if coin for a given symbol was not found.
+ * Helper function that goes through timestamped fiat rates returned from Coingecko and finds the closest one to the provided timestamp.
+ * @returns [timestamp, fiatRate] pair
+ */
+const findClosestTimestampValue = (timestamp: number, prices: Array<[number, number]>): number => {
```

Comment:
Would be nice to have tests for this

Reply:
tests added in [5ff6986](https://github.com/trezor/trezor-suite/pull/8642/commits/5ff69862c74f3c312467ee8cc0ec0d5d05a6a534)